lib/rack/query_parser.rb

```diff
@@ -94,20 +89,8 @@ def parse_query(qs, separator = nil, &unescaper)
     # Parses a query string by breaking it up at the '&', returning all key-value
     # pairs as an array of [key, value] arrays. Unlike parse_query, this preserves
     # all duplicate keys rather than collapsing them.
-    def parse_query_pairs(qs, separator = nil, &unescaper)
-      unescaper ||= method(:unescape)
-
-      pairs = []
-
-      check_query_string(qs, separator).split(separator ? (COMMON_SEP[separator] || /[#{separator}] */n) : DEFAULT_SEP).each do |p|
-        next if p.empty?
-        k, v = p.split('=', 2).map!(&unescaper)
-        pairs << [k, v]
-      end
-
-      return pairs
-    rescue ArgumentError => e
-      raise InvalidParameterError, e.message, e.backtrace
+    def parse_query_pairs(qs, separator = nil)
+      each_query_pair(qs, separator).to_a
```

Comment:
Related to the removal of `enum_for`:

```suggestion
      a = []
      each_query_pair(qs, separator) do |k, v|
        a << [k, v]
      end
      a
```